### PR TITLE
[feature flag] Enforce csrf protection on explore_json endpoint

### PR DESCRIPTION
**Pull request by graceguo-supercat**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
This PR is to resume the work in #7449. For some security concerns, we need to enforce CSRF protection on query request to `explore_json` endpoint.  

So I want to add a new feature flag: `ENABLE_EXPLORE_JSON_CSRF_PROTECTION`. When `ENABLE_EXPLORE_JSON_CSRF_PROTECTION` is set to true, user cannot make GET request to `explore_json`. 

The default value for this feature `False` (current behavior), explore_json accepts both GET and POST request.

### TEST PLAN
send GET request to `explore_json`, you will get `405 Method Not Allowed` exception.

### REVIEWERS
@DiggidyDave @betodealmeida   @john-bodley @mistercrunch

### CODE REVIEW

**Review by mistercrunch — APPROVED** [review]
LGMT

File: superset/config.py [hunk 1]
```diff
@@ -204,7 +204,8 @@
 # will result in combined feature flags of { 'FOO': True, 'BAR': True, 'BAZ': True }
 DEFAULT_FEATURE_FLAGS = {
     # Experimental feature introducing a client (browser) cache
-    "CLIENT_CACHE": False
+    "CLIENT_CACHE": False,
+    "ENABLE_EXPLORE_JSON_CSRF_PROTECTION": False,
```
Comment (etr2460) [inline superset/config.py]: Let's add a comment explaining this flag too?

**Review by etr2460 — APPROVED** [review]
one comment, otherwise lgtm

**Comment by graceguo-supercat** [comment]
I added a section in http://superset.incubator.apache.org/installation.html for featured flags:
Screen Shot 2019-07-29 at 12 18 38 PM

**Pull request merged**